server/api/region.go

```diff
@@ -262,14 +262,20 @@ func (h *regionsHandler) CheckRegionsReplicated(w http.ResponseWriter, r *http.R
 	}
 
 	regions := rc.ScanRegions(startKey, endKey, -1)
-	replicated := true
+	state := "REPLICATED"
 	for _, region := range regions {
 		if !opt.IsRegionReplicated(rc, region) {
-			replicated = false
+			state = "INPROGRESS"
+			for _, item := range rc.GetCoordinator().GetWaitingRegions() {
+				if item.Key == region.GetID() {
+					state = "WAITING"
+					break
+				}
+			}
 			break
 		}
 	}
-	h.rd.JSON(w, http.StatusOK, replicated)
+	h.rd.JSON(w, http.StatusOK, state)
```

Comment:
I think we can return a json object instead of the pure string for easy to extend in the future.

Reply:
It is actually JSON.